bitnami/spark/values-production.yaml

```diff
@@ -40,6 +40,14 @@ image:
 ## Spark master specific configuration
 ##
 master:
+  ## Metrics exporter pod Annotation and Labels
+  ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+  ##
+  #podAnnotations:
+  #  prometheus.io/path: "/metrics/applications/prometheus"
+  #  prometheus.io/scrape: "true"
+  #  prometheus.io/port: "8080"
```

Comment:
Just this is fine, as we do in other containers. Remember that pod annotations are not exclusive for metrics use (we are adding `metrics.podAnnotations` for that):

```  
## Additional pod annotations
## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
##
podAnnotations: {}
```

I would also move it to the end of the `master` object.

Reply:
fixed